src/slog.c

```diff
@@ -71,347 +81,352 @@ void slog_get_date(SlogDate *sdate)
-    fclose(file);
+    /* Cleanup. */
+    if (line) 
+    {
+        free(line);
+    }
+    fclose(fp);
 
     return ret;
 }
 
-
 /*
- * Retunr string in slog format. Function takes arguments 
- * and returns string in slog format without printing and 
- * saveing in file. Return value is char pointer.
+ * Generating string in form:
+ * yyyy.mm.dd-HH:MM:SS.UU - (some message).
  */
-char* slog_get(SlogDate *pDate, char *msg, ...) 
+char* slog_get(SlogDate *pDate, char *msg, ...)
 {
-    /* Used variables */
     static char output[MAXMSG];
     char string[MAXMSG];
 
-    /* Read args */
+    /* Read args. */
     va_list args;
     va_start(args, msg);
     vsprintf(string, msg, args);
     va_end(args);
 
-    /* Generate output string with date */
-    sprintf(output, "%02d.%02d.%02d-%02d:%02d:%02d.%02d - %s", 
-        pDate->year, pDate->mon, pDate->day, pDate->hour, 
-        pDate->min, pDate->sec, pDate->usec, string);
+    /* Generate output string with date. */
+    sprintf(output, "%02d.%02d.%02d-%02d:%02d:%02d.%02d - %s",
+            pDate->year, pDate->mon, pDate->day, pDate->hour,
+            pDate->min, pDate->sec, pDate->usec, string);
 
-    /* Return output */
     return output;
 }
 
-
 /*
- * slog - Log exiting process. Function takes arguments and saves
- * log in file if LOGTOFILE flag is enabled from config. Otherwise
- * it just prints log without saveing in file. Argument level is
- * logging level and flag is slog flags defined in slog.h header.
+ * Log exiting process. We save log in file
+ * if LOGTOFILE flag is enabled from config.
  */
 void slog(int level, int flag, const char *msg, ...)
 {
-    /* Lock for safe */
-    if (slg.td_safe) 
+    /* Lock thread for safe. */
+    if (slg.td_safe)
     {
         int rc;
         if ((rc = pthread_mutex_lock(&slog_mutex)))
         {
-            printf("<%s:%d> %s: [ERROR] Can not lock mutex: %s\n", 
-                __FILE__, __LINE__, __FUNCTION__, strerror(rc));
+            printf("[ERROR] <%s:%d> inside %s(): Can not lock mutex: %s\n",
+                   __FILE__, __LINE__, __func__, strerror(rc));
             exit(EXIT_FAILURE);
         }
     }
 
-    /* Used variables */
     SlogDate mdate;
     char string[MAXMSG];
     char prints[MAXMSG];
     char color[32], alarm[32];
     char *output;
 
     slog_get_date(&mdate);
+    /* Place zero-valued bytes. */
     bzero(string, sizeof(string));
     bzero(prints, sizeof(prints));
     bzero(color, sizeof(color));
     bzero(alarm, sizeof(alarm));
 
-    /* Read args */
+
+
+    /* Read args. */
     va_list args;
     va_start(args, msg);
     vsprintf(string, msg, args);
     va_end(args);
 
-    /* Check logging levels */
-    if(!level || level <= slg.level || level <= slg.file_level)
+
+    /* Check logging levels. */
+    if (!level || level <= slg.level || level <= slg.file_level)
     {
-        /* Handle flags */
-        switch(flag) 
+        /* Handle flags. */
+        switch (flag)
```

Comment:
In addition, ``SLOG_NONE`` and ``SLOG_FNONE`` both behave like ``default`` so we don't need the same code twice (Dry) and we save an additional if-claus (line [341](https://github.com/kala13x/slog/pull/12/files/1ea113f80999d38e13b984744ac6c2a172dfeed7#diff-0b7de98ff3b11d118c9550e81a75f340R341) and [360](https://github.com/kala13x/slog/pull/12/files/1ea113f80999d38e13b984744ac6c2a172dfeed7#diff-0b7de98ff3b11d118c9550e81a75f340R360)